modules/edge-ren/src/lib.rs

```diff
@@ -0,0 +1,464 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use codec::{Encode, Decode, HasCompact};
+use frame_support::{Parameter, decl_error, decl_event, decl_module, decl_storage, ensure, traits::{Get, EnsureOrigin, FungibleAsset, MintableAsset, BurnableAsset}};
+use frame_system::{ensure_none, ensure_signed};
+use edgeware_primitives::Balance;
+use sp_core::ecdsa;
+use sp_io::{crypto::secp256k1_ecdsa_recover, hashing::keccak_256};
+use sp_runtime::{
+	Permill,
+	ModuleId,
+	traits::{Member, StaticLookup, AccountIdConversion},
+	transaction_validity::{
+		InvalidTransaction, TransactionPriority, TransactionSource, TransactionValidity, ValidTransaction,
+	},
+	DispatchResult,
+};
+use sp_std::vec::Vec;
+use coinaddress as btc_address;
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+type EcdsaSignature = ecdsa::Signature;
+type DestAddress = Vec<u8>;
+
+
+
+type TokenIdOf<T> = <<T as Config>::Assets as FungibleAsset<<T as frame_system::Config>::AccountId>>::AssetId;
+type BalanceOf<T> = <<T as Config>::Assets as FungibleAsset<<T as frame_system::Config>::AccountId>>::Balance;
+
+const NAME_MAX_LENGTH : u8 = 32;
+
+pub trait Config: frame_system::Config {
+	type Event: From<Event<Self>> + Into<<Self as frame_system::Config>::Event>;
+	type RenVMTokenIdType: Member + Parameter + Default + Copy + HasCompact;
+	type RenvmBridgeUnsignedPriority: Get<TransactionPriority>;
+	type ControllerOrigin: EnsureOrigin<Self::Origin>;
+	type ModuleId: Get<ModuleId>;
+	type Assets: FungibleAsset<Self::AccountId> + MintableAsset<Self::AccountId> + BurnableAsset<Self::AccountId>;
```

Comment:
If the asset is burnable and mintable, then adding parameters to the struct to define whether it is seems pretty useless. Also all these assets will be burnable and mintable because that's how this bridge needs to operate.

Reply:
The mint and burn parameters in the RenTokenInfo struct are to temporarily disable minting/burning, without these we would have to delete the ren token from the registry (and also the asset from the assets if we are managing crud from the edge-ren pallet without a separate function to only delete the ren token from the registry), and reinstantiate it to disable and then enable the token.
If we see absolutely no case where we would need to halt minting/burning temporarily then I'll get rid of them.